sql/core/src/main/scala/org/apache/spark/sql/execution/ExpandExec.scala

```diff
@@ -42,9 +42,33 @@ case class ExpandExec(
   override lazy val metrics = Map(
     "numOutputRows" -> SQLMetrics.createMetric(sparkContext, "number of output rows"))
 
-  // The GroupExpressions can output data with arbitrary partitioning, so set it
-  // as UNKNOWN partitioning
```

Comment:
Shall we leverage `AliasAwareOutputPartitioning` and define the `outputExpressions` as the shared prefix of `projections`?

Reply:
I tried it, seems to work fine. Lets see if all other tests pass too.

Reply:
@cloud-fan Could you take another look at this